### PR TITLE
Revising documentation structure

### PR DESCRIPTION
**Pull request by pruksmhc**
This is a draft of how to revamp documentation, since we don't want to have too much duplicated information in many places. 
Right now the README consists of a mix of tutorial material and information on what is/is not supported by the framework. 
Thus, I suggest: 
-Adding tutorial material under tutorials folder and then pointing tutorials to those, thus removing a lot of the "Getting started" and "Adding a tasks" sections from README 
-Putting most of the Saving Preprocessed data, Pretrained embeddings, and documentation on what jiant supports to the website (where it's easier to read since there's a table of contents like format). 
-Leave what is absolutely essential (like git submodules and links to the documentation, as well as citation) in the README. 
Open to suggestions!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,10 +15,7 @@ A few things you might want to know about `jiant`:
 To run a simple example demo experiment using data from GLUE:
 
 1. Clone `jiant` and its submodules.
-2. Run the dependencies file.
-3. Download the GLUE data.
-4. Set up environment variables to point to your directories for data and experiment output.
-5. Run demo.sh.
+2. Follow this [Get Started tutorial]()!
```
Comment (sleepinyourhat) [inline README.md]: Cloning should be part of the tutorial, too! (So no more numbered list.)

File: README.md [hunk 2]
```diff
@@ -16,11 +16,7 @@ A few things you might want to know about `jiant`:
 
 To run a simple example demo experiment using data from GLUE:
 
-1. Clone `jiant` and its submodules.
-2. Run the dependencies file.
-3. Download the GLUE data.
-4. Set up environment variables to point to your directories for data and experiment output.
-5. Run demo.sh.
+Follow this [getting started tutorial]()!
```
Comment (sleepinyourhat) [inline README.md]: Fix.

File: README.md [hunk 3]
```diff
@@ -34,241 +30,10 @@ If you already cloned and just need to get the submodules, you can run:
...
-
-### Downloading data
-
-The repo contains a convenience python script for downloading all [GLUE](https://gluebenchmark.com/) data and standard splits.
-
-```
-python scripts/download_glue_data.py --data_dir data --tasks all
-```
-
-We also make use of many other data sources, including:
```
Comment (sleepinyourhat) [inline README.md]: This should stick around in some kind of documentation.

File: README.md [hunk 4]
```diff
@@ -34,241 +30,10 @@ If you already cloned and just need to get the submodules, you can run:
...
-- Image captioning: MSCOCO Dataset (http://cocodataset.org/#download). Specifically we use the following splits: 2017 Train images [118K/18GB], 2017 Val images [5K/1GB], 2017 Train/Val annotations [241MB].
-- Reddit: [reddit_comments dataset](https://bigquery.cloud.google.com/dataset/fh-bigquery:reddit_comments). Specifically we use the 2008 and 2009 tables.
-- DisSent: Details for preparing the corpora are in [`scripts/dissent/README`](scripts/dissent/README).
-- DNC (**D**iverse **N**atural Language Inference **C**ollection), i.e. recast data: The DNC is available [online](https://github.com/decompositional-semantics-initiative/DNC). Follow the instructions described there to download the DNC.
-- CCG: Details for preparing the corpora are in [`scripts/ccg/README`](scripts/ccg/README).
-- Edge probing analysis tasks: see _Papers_ below or [`probing/data`](probing/data/README.md) for more information.
-
-To incorporate the above data, placed the data in the data directory in its own directory (see task-directory relations in `src/preprocess.py` and `src/tasks.py`.
-
-### Running
```
Comment (sleepinyourhat) [inline README.md]: This seems worth keeping in the readme.

File: README.md [hunk 5]
```diff
@@ -16,11 +16,7 @@ A few things you might want to know about `jiant`:
...
 To run a simple example demo experiment using data from GLUE:
 
-1. Clone `jiant` and its submodules.
-2. Run the dependencies file.
-3. Download the GLUE data.
-4. Set up environment variables to point to your directories for data and experiment output.
-5. Run demo.sh.
+Follow this [getting started tutorial]()! 
 
 ### Submodules
```
Comment (sleepinyourhat) [inline README.md]: Probably fits in more in getting started.

File: README.md [hunk 6]
```diff
@@ -16,11 +16,7 @@ A few things you might want to know about `jiant`:
 
 To run a simple example demo experiment using data from GLUE:
```
Comment (sleepinyourhat) [inline README.md]: Emphasize that this is where the main setup instructions live.

File: README.md [hunk 7]
```diff
@@ -34,241 +30,10 @@ If you already cloned and just need to get the submodules, you can run:
...
-will run the demo config, but output to `$JIANT_PROJECT_PREFIX/my_exp/foobar`.
-
-To run the demo config, you will have to set environment variables. The best way to achieve that is to follow the instructions in [user_config_template.sh](user_config_template.sh)
-*  $JIANT_PROJECT_PREFIX: the where the outputs will be saved.
-*  $JIANT_DATA_DIR: location of the saved data. This is usually the location of the GLUE data in a simple default setup.
-*  $WORD_EMBS_FILE: location of any word embeddings you want to use (not necessary when using ELMo, GPT, or BERT). You can download GloVe (840B) [here](http://nlp.stanford.edu/data/glove.840B.300d.zip) or fastText (2M) [here](https://s3-us-west-1.amazonaws.com/fasttext-vectors/crawl-300d-2M.vec.zip).
-
-To have `user_config.sh` run automatically, follow instructions in [scripts/export_from_bash.sh](export_from_bash.sh). 
-
-## Command-Line Options
```
Comment (sleepinyourhat) [inline README.md]: Where does this go? Be careful not to delete any documentation...

File: README.md [hunk 8]
```diff
@@ -34,241 +30,10 @@ If you already cloned and just need to get the submodules, you can run:
...
-The trainer will write event data even if this flag is not used, and you can run Tensorboard separately as:
-```
-tensorboard --logdir <exp_dir>/<run_name>/tensorboard
-```
-
-`--notify <email_address>`: use this to enable notification emails via [SendGrid](https://sendgrid.com/). You'll need to make an account and set the `SENDGRID_API_KEY` environment variable to contain the (text of) the client secret key.
-
-`--remote_log` (or `-r`): use this to enable remote logging via Google Stackdriver. You can set up credentials and set the `GOOGLE_APPLICATION_CREDENTIALS` environment variable; see [Stackdriver Logging Client Libraries](https://cloud.google.com/logging/docs/reference/libraries#client-libraries-usage-python).
-
-## Saving Preprocessed Data
```
Comment (sleepinyourhat) [inline README.md]: Where does this go? Be careful not to delete any documentation...

File: README.md [hunk 9]
```diff
@@ -34,241 +30,10 @@ If you already cloned and just need to get the submodules, you can run:
...
-
-You should also set ``data_dir`` and  ``word_embs_file`` options to point to the directories containing the data (e.g. the output of the ``scripts/download_glue_data`` script) and word embeddings (optional, not needed when using ELMo, see later sections) respectively.
-
-To force rereading and reloading of the tasks, perhaps because you changed the format or preprocessing of a task, delete the objects in the directories named for the tasks (e.g., `QQP/`) or use the option ``reload_tasks = 1``.
-
-To force rebuilding of the vocabulary, perhaps because you want to include vocabulary for more tasks, delete the objects in `vocab/` or use the option ``reload_vocab = 1``.
-
-To force reindexing of a task's data, delete some or all of the objects in `preproc/` or use the option ``reload_index = 1`` and set ``reindex_tasks`` to the names of the tasks to be reindexed, e.g. ``reindex_tasks=\"sst,mnli\"``. You should do this whenever you rebuild the task objects or vocabularies.
-
-## Models (and how to add a Model)
```
Comment (sleepinyourhat) [inline README.md]: Where does this go? Be careful not to delete any documentation...

File: README.md [hunk 10]
```diff
@@ -34,241 +30,10 @@ If you already cloned and just need to get the submodules, you can run:
...
-
-	* Single sentence classification tasks
-	* Pair sentence classification tasks
-	* Regression tasks
-	* Tagging tasks
-	* Span classification Tasks 
-		* To run these, we currently require an extra preprocessing step, which consists of preprocessing the data to get BERT tokenized span indices. SpanTasks expects the files to be in `json` format and be named as `{file_name}.retokenized.{tokenizer_name}`.
-	* seq2seq tasks (partial/tentative support only) partially supported.
-
-### GPT, BERT, and Transformers
```
Comment (sleepinyourhat) [inline README.md]: Probably best to keep in the readme?

File: README.md [hunk 11]
```diff
@@ -34,241 +30,10 @@ If you already cloned and just need to get the submodules, you can run:
...
-
-When using BERT, we follow the procedures set out in the original work as closely as possible: For pair sentence tasks, we concatenate the sentences with a special `[SEP]` token. Rather than max-pooling, we take the first representation of the sequence (corresponding to the special `[CLS]` token) as the representation of the entire sequence.
-We also have support for the version of Adam that was used in training BERT (``optimizer = bert_adam``).
-
-We also include an experimental option to use a shared [Transformer](https://arxiv.org/abs/1706.03762) in place of the shared BiLSTM by setting ``sent_enc = transformer``. When using a Transformer, we use the [Noam learning rate scheduler](https://github.com/allenai/allennlp/blob/master/allennlp/training/learning_rate_schedulers.py#L84), as that seems important to training the Transformer thoroughly. 
-
-## Trainer
```
Comment (sleepinyourhat) [inline README.md]: Where does this go? Be careful not to delete any documentation...

File: README.md [hunk 12]
```diff
@@ -34,241 +30,10 @@ If you already cloned and just need to get the submodules, you can run:
...
-For example, ``pretrain_tasks = \"sst,mnli,foo\", target_tasks = \"qnli,bar,sst,mnli,foo\"`` (HOCON notation requires escaped quotes in command line arguments).
-Note: if you want to train and evaluate on a task, that task must be in both ``pretrain_tasks`` and ``target_tasks``.
-
-We support two modes of adapting pretrained models to target tasks. 
-Setting `transfer_paradigm = finetune` will fine-tune the entire model while training for a target task.
-The mode will create a copy of the model _per target task_.
-Setting `transfer_paradigm = frozen` will only train the target-task specific components while training for a target task.
-If using ELMo and `sep_embs_for_skip = 1`, we will also learn a task-specific set of layer-mixing weights.
-
-## Adding New Tasks
```
Comment (sleepinyourhat) [inline README.md]: Replace with pointer?

File: README.md [hunk 13]
```diff
@@ -34,241 +30,10 @@ If you already cloned and just need to get the submodules, you can run:
...
-    * Your task is correctly being handled in ``forward()`` of ``MultiTaskModel``. The model will receive the task class you created and a batch of data, where each batch is a dictionary with keys of the ``Instance`` objects you created in preprocessing, as well as a ``predict`` flag that indicates if your forward function should generate predictions or not.
-    * You create additional methods or add branches to existing methods as necessary. If you do add additional methods, make sure to make use of the ``sent_encoder`` attribute of the model, which is shared amongst all tasks.
-
-Note: The current training procedure is task-agnostic: we randomly sample a task to train on, pass a batch to the model, and receive an output dictionary at least containing a ``loss`` key. Training loss should be calculated within the model; validation metrics should also be computed within AllenNLP ``scorer``s and not in the training loop. So you should *not* need to modify the training loop; please reach out if you think you need to.
-
-Feel free to create a pull request to add an additional task if you expect that it'll be useful to others.
-
-## Pretrained Embeddings
```
Comment (sleepinyourhat) [inline README.md]: Where does this go? Be careful not to delete any documentation...

File: README.md [hunk 14]
```diff
@@ -305,10 +70,26 @@ Post an issue here on GitHub if you have any problems, and create a pull request
...
+- First run: pretrain on task set A
+   - pretrain_tasks=“task_a1,task_a2”, target_tasks=“”
+- Second run: load checkpoints, and train on task set B and then C:
+   - load_model = 1
+   - load_target_train_checkpoint_arg=/path/to/saved/run
+   - pretrain_tasks=“task_b1,task_b2, target_tasks=task_c1,task_c2”
+
+***How can I pass BERT embeddings straight to the classifier without a sentence encoder? ***
+
+Right now, you need to set skip_embs=1 and sep_embs_for_skip=1 just because of the current way
```
Comment (sleepinyourhat) [inline README.md]: Use backticks.

File: tutorials/setup_tutorial.md [hunk 15]
```diff
@@ -0,0 +1,253 @@
...
+before running any `jiant` code. (To deactivate run: `source deactivate`)
+
+Some requirements may only be needed for specific configurations. If you have trouble installing a specific dependency and suspect that it isn't needed for your use case, create an issue or a pull request, and we'll help you get by without it.
+
+You will also need to install dependencies for `nltk` if you do not already have them:
+
+```
+python -m nltk.downloader -d  perluniprops nonbreaking_prefixes punkt
+``` 
+***SB: This still fills my jiant directory with junk... not ideal. Either look into the suggested nltk behavior (or as a very, very hacky marginal workaround, add this to the shared .gitignore).***
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: Fix these. (Search for ***.)

File: tutorials/setup_tutorial.md [hunk 16]
```diff
@@ -0,0 +1,253 @@
...
+* `params.conf` (a saved version of the parameters used)
+* written predictions for test for each of the target trained tasks (with file names {task_name}-test.tsv
+* a saved checkpoint of your best validation metric.
+
+Additionally, the validation scores will be written in `results.tsv` in your experiment directory with the name of the run it belongs to. 
+
+And there you have it! Your first experiment.
+If you are looking for where to go next, check out our documentation [here](https://jiant.info/documentation/#/?id=saving-preprocessed-data)!
+
+Happy jianting!
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: `jiant`ing

File: README.md (hunk 13 above)
Comment (pruksmhc) [reply README.md]: It goes here! 
https://jiant.info/documentation#/?id=supported-pretrained-embeddings

File: README.md (hunk 3 above)
Comment (pruksmhc) [reply README.md]: Yes, it is present in the Getting Started tutorial as well as here: https://jiant.info/documentation#/?id=data-sources

File: README.md (hunk 7 above)
Comment (pruksmhc) [reply README.md]: Again, this is in https://jiant.info/documentation#/?id=command-line-options

File: README.md (hunk 10 above)
Comment (pruksmhc) [reply README.md]: My concern is that the README will get too long (as opposed to having it in a documentation site).

File: README.md (hunk 6 above)
Comment (sleepinyourhat) [reply README.md]: It seems very odd to leave this as multiple lines/sentences, but we can clean this up later if needed.

File: README.md [hunk 17]
```diff
@@ -16,258 +16,26 @@ A few things you might want to know about `jiant`:
...
-- Edge probing analysis tasks: see _Papers_ below or [`probing/data`](probing/data/README.md) for more information.
-
-To incorporate the above data, placed the data in the data directory in its own directory (see task-directory relations in `src/preprocess.py` and `src/tasks.py`.
-
-### Running
-
-To run an experiment, make a config file similar to `config/demo.conf` with your model configuration. You can use the `--overrides` flag to override specific variables. For example:
-```sh
+## Running
+To run an experiment, make a config file similar to `config/demo.conf` with your model configuration. You can use the `--overrides` flag to override specific variables. For example:
```
Comment (sleepinyourhat) [inline README.md]: _In addition,_ you can use ...

File: tutorials/adding_tasks.md [hunk 18]
```diff
@@ -0,0 +1,172 @@
+# How to Add a Task
```
Comment (sleepinyourhat) [inline tutorials/adding_tasks.md]: Add backticks for code, with links where possible.

File: tutorials/setup_tutorial.md [hunk 19]
```diff
@@ -0,0 +1,256 @@
...
+
+We also support quite a few other data sources (check [here](https://jiant.info/documentation#/?id=data-sources)  for a list).
+
+
+Finally, you'll need to set a few environment variables in [user_config_template.sh](https://github.com/nyu-mll/jiant/blob/master/user_config_template.sh), which include:
+
+
+* $JIANT_PROJECT_PREFIX: the directory where things like logs and model checkpoints will be saved.
+* $JIANT_DATA_DIR: location of the data you want to train and evaluate on. As a starting point, this is often the directory created by the GLUE or SuperGLUE data downloaders. Let's use the `data/` directory for GLUE for now. 
+* $WORD_EMBS_FILE: location of any word embeddings you want to use (not necessary when using ELMo, GPT, or BERT). You can download GloVe (840B) [here]() or fastText (2M) [here]().
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: Fix.

File: README.md (hunk 2 above)
Comment (pruksmhc) [reply README.md]: I'll put in the official link once the tutorials are checked in

File: tutorials/setup_tutorial.md [hunk 20]
```diff
@@ -0,0 +1,255 @@
...
+* a checkpoint of the best model state (based on your scores)
+* a `log.log` file which contains all the logs
+* `params.conf` (a saved version of the parameters used)
+* written predictions for test for each of the target trained tasks (with file names {task_name}-test.tsv
+* a saved checkpoint of your best validation metric.
+
+Additionally, the validation scores will be written in `results.tsv` in your experiment directory with the name of the run it belongs to. 
+
+And there you have it! Your first experiment.
+If you are looking for where to go next, check out our documentation [here](https://jiant.info/documentation/#/?id=saving-preprocessed-data)!
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: Remove ?id bit.

File: tutorials/setup_tutorial.md [hunk 21]
```diff
@@ -0,0 +1,255 @@
...
+from the root of the `jiant` directory, which adds the source command directly to your machine's bash file.
+
+Now that we've set up the environment, let's get started!
+
+## 3. Running our first experiment 
+
+### 3.a) Configuring our experiment
+
+Here, we'll try pretraining in a multitask setting on SST and MRPC and then finetuning on STS-B and WNLI separately using a BiLSTM sentence encoder and word embeddings trained from scratch. 
+This is almost exactly what is specified in `config/demo.conf`, with one major change. From here, we suggest you to go to [`config/demo.conf`](https://github.com/nyu-mll/jiant/blob/master/config/demo.conf), make a copy in config/tutorial.conf, and follow along - we'll explain everything that is in that file in a bit.
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: copy 'called' config/...
and use `` for the path.

File: tutorials/setup_tutorial.md [hunk 22]
```diff
@@ -0,0 +1,255 @@
+# Tutorial: Getting Started 
+
+
+Wecome to `jiant`! Let's help get you set up and running a demo experiment!
+
+## 1. Install
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: You skip from 1 to 3. Just delete the numbers?

File: README.md [hunk 23]
```diff
@@ -14,260 +14,26 @@ A few things you might want to know about `jiant`:
 
 ## Getting Started
 
-To run a simple example demo experiment using data from GLUE:
+To find the setup instructions for using jiant and to run a simple example demo experiment using data from GLUE, follow this [getting started tutorial]()!
```
Comment (sleepinyourhat) [inline README.md]: You should be able to work out the URL now before merging.

File: tutorials/setup_tutorial.md (hunk 22 above)
Comment (pruksmhc) [reply tutorials/setup_tutorial.md]: I do have 2 below

Comment (sleepinyourhat) [reply tutorials/setup_tutorial.md]: Oops—sorry. Indentation threw me off. (But won't be rendered, I assume...)

File: README.md [hunk 24]
```diff
@@ -14,260 +14,26 @@ A few things you might want to know about `jiant`:
...
-
-The base model class is a MultiTaskModel. To add another model, first add the class of the model to modules/modules.py, and then add the model construction in ``make_sent_encoder()`` (called in ``build_model()``) in src/models.py.
+    --overrides "exp_name = my_exp, run_name = foobar, d_hid = 256"	
+```	
+will run the demo config, but output to `$JIANT_PROJECT_PREFIX/my_exp/foobar`.	
+ To run the demo config, you will have to set environment variables. The best way to achieve that is to follow the instructions in [user_config_template.sh](user_config_template.sh)	
+*  $JIANT_PROJECT_PREFIX: the where the outputs will be saved.	
+*  $JIANT_DATA_DIR: location of the saved data. This is usually the location of the GLUE data in a simple default setup.	
+*  $WORD_EMBS_FILE: location of any word embeddings you want to use (not necessary when using ELMo, GPT, or BERT). You can download GloVe (840B) [here](http://nlp.stanford.edu/data/glove.840B.300d.zip) or fastText (2M) [here](https://s3-us-west-1.amazonaws.com/fasttext-vectors/crawl-300d-2M.vec.zip)..
```
Comment (sleepinyourhat) [inline README.md]: .. at end of line.

File: README.md [hunk 25]
```diff
@@ -303,12 +69,6 @@ This package is released under the [MIT License](LICENSE.md). The material in th
 
 Post an issue here on GitHub if you have any problems, and create a pull request if you make any improvements (substantial or cosmetic) to the code that you're willing to share.
 
-## FAQs
```
Comment (sleepinyourhat) [inline README.md]: Assuming this did in fact move somewhere...

File: tutorials/setup_tutorial.md [hunk 26]
```diff
@@ -0,0 +1,255 @@
...
+05/03 09:59:23 AM: micro_avg: 0.168, macro_avg: 0.356, sts-b_corr: 0.149, sts-b_pearsonr: 0.146, sts-b_spearmanr: 0.152, wnli_accuracy: 0.563
+05/03 09:59:23 AM: Done!
+```
+
+After running this experiment, you should have in your run directory:
+
+* a checkpoint of the best model state (based on your scores)
+* a `log.log` file which contains all the logs
+* `params.conf` (a saved version of the parameters used)
+* written predictions for test for each of the target trained tasks (with file names {task_name}-test.tsv
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: broken parens/formatting

File: tutorials/setup_tutorial.md [hunk 27]
```diff
@@ -0,0 +1,255 @@
...
+
+The logs include:
+
+* The process of setting up and loading tasks
+* Restoring checkpoints (if applicable)
+* Printing out the Model architecture
+* Indexing your data into AllenNLP instances for preparation for training.
+* When you get to training stage, updates of the current progress and validation. 
+
+One important thing to notice is that during training, the updates will swap between sst and mrpc. This is because for each training batch, we sample the tasks based on a parameter you can set in your experiment `weighting_method`, which is automatically set to proportional (so the largerthetask, the larger the probablty it will get sampled for a batch).
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: thelargerthe

File: tutorials/setup_tutorial.md [hunk 28]
```diff
@@ -0,0 +1,255 @@
...
+python main.py --config_file config/tutorial.conf \
+    --overrides "exp_name = my_exp, run_name = foobar"
+```
+
+will run the demo config, but write output to `$JIANT_PROJECT_PREFIX/my_exp/foobar`.
+
+
+### 3.b) Understanding the output logs
+
+We do support Tensorboard, however, you can also look at the logs to make sure everything in your experiment is running smoothly.
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: do support => support

File: tutorials/setup_tutorial.md [hunk 29]
```diff
@@ -0,0 +1,255 @@
...
+```
+
+will run the demo config, but write output to `$JIANT_PROJECT_PREFIX/my_exp/foobar`.
+
+
+### 3.b) Understanding the output logs
+
+We do support Tensorboard, however, you can also look at the logs to make sure everything in your experiment is running smoothly. 
+
+The logs include:
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: "There's a lot going on here (including some debugging information that we're working on suppressing), but a lot of it is useful:"

File: tutorials/setup_tutorial.md [hunk 30]
```diff
@@ -0,0 +1,255 @@
...
+
+* The process of setting up and loading tasks
+* Restoring checkpoints (if applicable)
+* Printing out the Model architecture
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: Model => model

File: tutorials/setup_tutorial.md [hunk 31]
```diff
@@ -0,0 +1,255 @@
...
+* Indexing your data into AllenNLP instances for preparation for training.
+* When you get to training stage, updates of the current progress and validation.
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: Make sure these are in order.

File: tutorials/setup_tutorial.md [hunk 32]
```diff
@@ -0,0 +1,255 @@
...
+
+We also support quite a few other data sources (check [here](https://jiant.info/documentation#/?id=data-sources)  for a list).
+
+
+Finally, you'll need to set a few environment variables in [user_config_template.sh](https://github.com/nyu-mll/jiant/blob/master/user_config_template.sh), which include:
+
+
+* $JIANT_PROJECT_PREFIX: the directory where things like logs and model checkpoints will be saved.
+* $JIANT_DATA_DIR: location of the data you want to train and evaluate on. As a starting point, this is often the directory created by the GLUE or SuperGLUE data downloaders. Let's use the `data/` directory for GLUE for now. 
+* $WORD_EMBS_FILE: location of any word embeddings you want to use (not necessary when using ELMo, GPT, or BERT). You can download GloVe (840B) [here](http://nlp.stanford.edu/data/glove.840B.300d.zip) or fastText (2M) [here](https://s3-us-west-1.amazonaws.com/fasttext-vectors/crawl-300d-2M.vec.zip)
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: final period missing.

File: tutorials/setup_tutorial.md [hunk 33]
```diff
@@ -0,0 +1,255 @@
...
+python scripts/download_glue_data.py --data_dir data --tasks all
+```
+
+We also support quite a few other data sources (check [here](https://jiant.info/documentation#/?id=data-sources)  for a list).
+
+
+Finally, you'll need to set a few environment variables in [user_config_template.sh](https://github.com/nyu-mll/jiant/blob/master/user_config_template.sh), which include:
+
+
+* $JIANT_PROJECT_PREFIX: the directory where things like logs and model checkpoints will be saved.
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: `` for these var names.

File: tutorials/setup_tutorial.md [hunk 34]
```diff
@@ -0,0 +1,255 @@
...
+* $JIANT_DATA_DIR: location of the data you want to train and evaluate on. As a starting point, this is often the directory created by the GLUE or SuperGLUE data downloaders. Let's use the `data/` directory for GLUE for now. 
+* $WORD_EMBS_FILE: location of any word embeddings you want to use (not necessary when using ELMo, GPT, or BERT). You can download GloVe (840B) [here](http://nlp.stanford.edu/data/glove.840B.300d.zip) or fastText (2M) [here](https://s3-us-west-1.amazonaws.com/fasttext-vectors/crawl-300d-2M.vec.zip)
+
+
+To avoid having your custom paths overwritten by future updates, you should save a copy of this file as `user_config.sh` (or something similar, but `user_config.sh` will be automatically ignored by git.
```
Comment (sleepinyourhat) [inline tutorials/setup_tutorial.md]: Fix parens.
'... a file with the name `user_config.sh` will be ...'

File: README.md (hunk 25 above)
Comment (pruksmhc) [reply README.md]: Yes

**Pull request merged**